hazelcast/src/main/java/com/hazelcast/config/RestApiConfig.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2008-2019, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * This class allows to control which parts of Hazelcast REST API will be enabled. There are 2 levels of control:
+ * <ul>
+ * <li>overall REST access (enabled by default);</li>
+ * <li>access to REST endpoint groups (see {@link RestEndpointGroup}).</li>
+ * </ul>
+ */
+public class RestApiConfig {
+
+    private boolean enabled;
+
+    private final Set<RestEndpointGroup> enabledGroups = Collections
```

Comment:
`EnumSet` should be a better fit for this data-type (for concurrency see. `Collections.synchronized(...)`)

Reply:
:+1: Nice catch, I'll change the impl to an `EnumSet`.